lib/data-types.js

```diff
@@ -39,24 +42,47 @@ var util = require('util')
  */
 
 var ABSTRACT = function(options) {
-
 };
 
 ABSTRACT.prototype.dialectTypes = '';
 
-ABSTRACT.prototype.toString = function() {
-  return this.toSql();
+ABSTRACT.prototype.toString = function(options) {
+  return this.toSql(options);
 };
 ABSTRACT.prototype.toSql = function() {
   return this.key;
 };
-ABSTRACT.prototype.warn = function(text) {
-  if (!warnings[text]) {
-    warnings[text] = true;
-    console.warn('>> WARNING:', text, '\n>> Check:', this.dialectTypes);
+ABSTRACT.warn = function(link, text) {
+  deprecate(text + '\n>> Check: ' + link);
```

Comment:
This prints a deprecation notice but it's not really deprecated. It's just not the right type used. And the line provided by the module is the one from the DataTypes but it should be the line where the user inserted the wrong type. So this should probably be reconsidered.